coralogix/resource_coralogix_dashboard.go

```diff
@@ -0,0 +1,1663 @@
+		sourceMap = map[string]interface{}{
+			"metric_label": metricLabel,
+		}
+	case *dashboardv1.MultiSelect_Source_ConstantList:
+		constantList := flattenConstantListSource(sourceValue.ConstantList)
+		sourceMap = map[string]interface{}{
+			"constant_list": constantList,
+		}
+	}
+	return []interface{}{
+		sourceMap,
+	}
+}
+
+func flattenLogPathSource(logPath *dashboardv1.MultiSelect_LogsPathSource) interface{} {
+	value := logPath.GetValue().GetValue()
+	return []interface{}{
+		map[string]interface{}{
+			"value": value,
+		},
+	}
+}
+
+func flattenMetricLabelSource(metricLabel *dashboardv1.MultiSelect_MetricLabelSource) interface{} {
+	metricName := metricLabel.GetMetricName().GetValue()
+	label := metricLabel.GetLabel().GetValue()
+	return []interface{}{
+		map[string]interface{}{
+			"metric_name": metricName,
+			"label":       label,
+		},
+	}
+}
+
+func flattenConstantListSource(constantList *dashboardv1.MultiSelect_ConstantListSource) interface{} {
+	values := wrappedStringSliceToStringSlice(constantList.GetValues())
+	return []interface{}{
+		map[string]interface{}{
+			"values": values,
+		},
+	}
+}
+
+func resourceCoralogixDashboardUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	dashboard, diags := extractDashboard(d)
+	if diags != nil {
+		return diags
+	}
+	updateDashboardRequest := &dashboardv1.ReplaceDashboardRequest{
+		Dashboard: dashboard,
+	}
+
+	log.Printf("[INFO] Updating dashboard: %#v", updateDashboardRequest)
+	DashboardResp, err := meta.(*clientset.ClientSet).Dashboards().UpdateDashboard(ctx, updateDashboardRequest)
+	if err != nil {
+		log.Printf("[ERROR] Received error: %#v", err)
+		return handleRpcError(err, "dashboard")
+	}
+
+	Dashboard := DashboardResp.ProtoReflect()
+	log.Printf("[INFO] Submitted updated dashboard: %#v", Dashboard)
+	d.SetId(updateDashboardRequest.GetDashboard().GetId().GetValue())
+
+	return resourceCoralogixDashboardRead(ctx, d, meta)
+}
+
+func resourceCoralogixDashboardDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	id := d.Id()
+	log.Printf("[INFO] Deleting dashboard %s\n", id)
+	deleteAlertRequest := &dashboardv1.DeleteDashboardRequest{DashboardId: &dashboardv1.UUID{Value: id}}
+	_, err := meta.(*clientset.ClientSet).Dashboards().DeleteDashboard(ctx, deleteAlertRequest)
+	if err != nil {
+		log.Printf("[ERROR] Received error: %#v\n", err)
+		return handleRpcErrorWithID(err, "dashboard", id)
+	}
+	log.Printf("[INFO] dashboard %s deleted\n", id)
+
+	d.SetId("")
+	return nil
+}
+
+func DashboardSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			Description:  "Dashboard name.",
+		},
+		"description": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Dashboard description.",
+		},
+		"layout": {
```

Comment:
We need to be careful here, if the schema is breaking due to a change we did, we need to make sure we'll have a way to support backward compatible.

Reply:
Also, don't we want to support file structure like we did in the enrichment (e.g. file = <layout.json>)